pkg/agent/controller/supportbundlecollection/support_bundle_controller.go

```diff
@@ -0,0 +1,357 @@
+)
+
+var (
+	emptyWatch      = watch.NewEmptyWatch()
+	defaultFS       = afero.NewOsFs()
+	defaultExecutor = exec.New()
+	// Declared as variable for testing.
+	newAgentDumper = support.NewAgentDumper
+)
+
+type SupportBundleController struct {
+	nodeName              string
+	nodeType              string
+	externalNodeNamespace string
+	antreaClientGetter    agent.AntreaClientProvider
+	queue                 *workqueue.Type
+	ovsCtlClient          ovsctl.OVSCtlClient
+	aq                    agentquerier.AgentQuerier
+	npq                   querier.AgentNetworkPolicyInfoQuerier
+	v4Enabled             bool
+	v6Enabled             bool
+}
+
+func NewSupportBundleController(nodeName string, nodeType string, externalNodeNamespace string, antreaClientGetter agent.AntreaClientProvider, ovsCtlClient ovsctl.OVSCtlClient,
+	aq agentquerier.AgentQuerier, npq querier.AgentNetworkPolicyInfoQuerier, v4Enabled, v6Enabled bool) *SupportBundleController {
+	c := &SupportBundleController{
+		nodeName:              nodeName,
+		nodeType:              nodeType,
+		externalNodeNamespace: externalNodeNamespace,
+		antreaClientGetter:    antreaClientGetter,
+
+		queue:        workqueue.NewNamed("supportbundle"),
+		ovsCtlClient: ovsCtlClient,
+		aq:           aq,
+		npq:          npq,
+		v4Enabled:    v4Enabled,
+		v6Enabled:    v6Enabled,
+	}
+	return c
+}
+func (c *SupportBundleController) watchSupportBundleCollections() {
+	klog.Info("Starting watch for SupportBundleCollections")
+	antreaClient, err := c.antreaClientGetter.GetAntreaClient()
+	if err != nil {
+		klog.Warningf("Failed to get antrea client: %v", err)
+		return
+	}
+	options := metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("nodeName", c.nodeName).String(),
+	}
+	watcher, err := antreaClient.ControlplaneV1beta2().SupportBundleCollections().Watch(context.TODO(), options)
+	if err != nil {
+		klog.Warningf("Failed to start watch for SupportBundleCollections: %v", err)
+		return
+	}
+	// Watch method doesn't return error but "emptyWatch" in case of some partial data errors,
+	// e.g. timeout error. Make sure that watcher is not empty and log warning otherwise.
+	if reflect.TypeOf(watcher) == reflect.TypeOf(emptyWatch) {
+		klog.Warning("Failed to start watch for SupportBundleCollections, please ensure antrea service is reachable for the agent")
+		return
+	}
+
+	klog.Info("Started watch for SupportBundleCollections")
+	eventCount := 0
+	defer func() {
+		klog.Infof("Stopped watch for SupportBundleCollections, total items received: %d", eventCount)
+		watcher.Stop()
+	}()
+
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return
+			}
+			switch event.Type {
+			case watch.Bookmark:
+				klog.Infof("Received Bookmark event")
+			case watch.Added:
+				c.enqueueSupportBundleCollection(event.Object.(*cpv1b2.SupportBundleCollection))
+				klog.Infof("Enqueue added SupportBundleCollections (%#v)", event.Object)
+			case watch.Modified:
```

Comment:
Maybe remove this "modified" case, as we do not update the internal support bundle collection object in Controller.